lib/shared/sdk/usbboot/index.js

```diff
@@ -251,6 +251,21 @@ class USBBootAdapter extends EventEmitter {
     usb.listDevices().filter(isUsbBootCapableUSBDevice).map((device) => {
     /* eslint-enable lodash/prefer-lodash-method */
 
+      if (!device.driver) {
+        return {
+          device: `${usbIdToString(device.deviceDescriptor.idVendor)}:${usbIdToString(device.deviceDescriptor.idProduct)}`,
```

Comment:
So, is this the first time that a "drivelist object" is coming directly from the SDK rather than from `drivelist` itself? I guess other SDK modues might also need to do something similar in future, so we should probably document some kind of "drivelist object spec" with what compulsory and optional properties (and their types) we expect this object to have?
Is there code in Etcher that emits a warning if a drivelist object is missing some of its compulsory properties? (and then presumably just ignores that incomplete drivelist object)

Reply:
@lurch We've been already doing so for a while in the usbboot implementation (look in this same file). The required properties should be defined by the SDK (@jhermsmeier is working on the spec). Drivelist happens to be an implementation that adheres to the current required properties.

> Is there code in Etcher that emits a warning if a drivelist object is missing some of its compulsory properties? (and then presumably just ignores that incomplete drivelist object)

Most of the functions have fallbacks in case a certain property is not defined. In any case, this will probably be discussed in much more detail in the SDK spec.